Sources/SotoCore/Credential/ConfigFileCredentialProvider.swift

```diff
@@ -134,7 +134,16 @@ struct AWSConfigFileCredentialProvider: CredentialProvider {
             return pth
         }
         #else
+        #if os(macOS)
+        guard let home = getpwuid(getuid())?.pointee.pw_dir else {
+            return filePath
+        }
+        let homePath = FileManager.default.string(withFileSystemRepresentation: home, length: Int(strlen(home)))
+        let expandedPath = filePath.replacingOccurrences(of: "~", with: homePath)
```

Comment:
Any reason we can't use `String(cString: home)` instead of `FileManager.default.string(withFileSystemRepresentation:)` here?

Reply:
no reason, ignorance on my side :-) 
What's the reason your pushing to not use FileManager ? Performance ? Binaries size ?

Reply:
change is done

Reply:
Limiting our exposure to Foundation, although in this situation it isn't such an issue as this is only running on macOS.